### PR TITLE
Add connector_office_365_hr_leave to push leaves to o365

### PR DESCRIPTION
**Pull request by guewen**
It contains the same subset of commits than #77, which needs to be merged first.
I separated the new addon to ease review of #77, but I can do one PR if you prefer.
Only the last commit is relevant here.

This addon pushes the events created for HR leaves to office365.

### CODE REVIEW

File: connector_office_365/models/calendar_event.py [hunk 1]
```diff
@@ -142,24 +207,24 @@ def office_365_open(self):
             'url': self.office_365_url
         }
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        records = super(CalendarEvent, self).create(vals_list)
```
Comment (Tonow-c2c) [inline connector_office_365/models/calendar_event.py]:
```suggestion
        records = super().create(vals_list)
```

no blocking this just python 3

File: connector_office_365/models/res_users.py [hunk 2]
```diff
@@ -163,7 +169,7 @@ def office_365_request(self, method, url, data=None, headers=None):
                                    data=data,
                                    client_id=client_id,
                                    client_secret=client_secret)
-        if response.status_code not in [200, 204]:
+        if not response.ok:
```
Comment (Tonow-c2c) [inline connector_office_365/models/res_users.py:184]: ok is just [200, 204] or it's wider ?

File: connector_office_365_hr_leave/views/hr_leave_views.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  <record id="hr_leave_view_form_o365" model="ir.ui.view">
+    <field name="name">hr.leave.view.form.o365</field>
+    <field name="model">hr.leave</field>
+    <field name="inherit_id" ref="hr_holidays.hr_leave_view_form"/>
+    <field name="arch" type="xml">
+      <field name="state" position="before">
+        <field name="need_o365_authenticate" invisible="1"/>
+        <field name="need_o365_manual_push" invisible="1"/>
+        <button string="Office 365 authentication"
```
Comment (Tonow-c2c) [inline connector_office_365_hr_leave/views/hr_leave_views.xml:12]: Any translation ?

File: connector_office_365/models/res_users.py (hunk 2 above)
Comment (guewen) [reply connector_office_365/models/res_users.py:184]: Read the commit's message https://github.com/onesteinbv/addons-onestein/pull/78/commits/a61b58eba1f3b69d6af0765c45209b5436a3da9f

Yes, it is wider. It should cover all 2xx and 3xx codes.

Comment (Tonow-c2c) [reply connector_office_365/models/res_users.py:184]: ok thanks :+1:

**Review by Tonow-c2c — APPROVED** [review]
Just code review

**Comment by guewen** [comment]
Rebased

**Review by rafaelbn — APPROVED** [review]

**Comment by tarteo** [comment]
I'm sorry for this late response. Could you resolve the conflicts? Then I'll merge it right away.

**Comment by jcoux** [comment]
@tarteo PR rebased ;)

**Comment by tarteo** [comment]
Thanks!

**Pull request merged**